libgnucash/engine/Account.cpp

```diff
@@ -4088,6 +4138,18 @@ xaccAccountSetPlaceholder (Account *acc, gboolean val)
     set_boolean_key(acc, {"placeholder"}, val);
 }
 
+gboolean
+xaccAccountGetOpeningBalance (const Account *acc)
+{
+    return boolean_from_key(acc, {"opening-balance"});
```

Comment:
KVP lookups are expensive and this is going to get called a lot when searching for the opening balance account. You can speed up the check a bit by
```
       return GET_PRIVATE(acc)->type == ACCT_TYPE_EQUITY && boolean_from_key(acc, {"opening-balance"});
```